package p05.throw_exception;

public class ThrowExceptionEx {

	public static void main(String[] args) throws ClassNotFoundException {
		
//		try {
			findClass();
//			System.out.println("정상 처리됨");
//		} catch (ClassNotFoundException e) {
//			System.out.println("compile-time Exception 발생");
//		}
	}

	// method 선언문에서 throws ClassNotFoundException 의미 
	// - method 내부에서 ClassNotFoundException이 발생이 되면 method를 호출한 method로 Exception을 넘김 (throw)
	public static void findClass() throws ClassNotFoundException {
		
		Class cls = Class.forName("java.lang.String2");
	}
}
